src/app/core/common-components/entity-select/entity-select.component.ts

```diff
@@ -88,11 +92,33 @@ export class EntitySelectComponent<E extends Entity> implements OnChanges {
         untilDestroyed(this),
         filter((isLoading) => !isLoading),
       )
-      .subscribe((_) => {
-        this.selectedEntities = this.allEntities.filter((e) =>
-          sel.find((s) => s === e.getId(true) || s === e.getId()),
-        );
-      });
+      .subscribe(() => this.initSelectedEntities(sel));
+  }
+
+  private async initSelectedEntities(selected: string[]) {
+    const entities: E[] = [];
+    for (const s of selected) {
+      const found = this.allEntities.find(
+        (e) => s === e.getId(true) || s === e.getId(),
+      );
+      if (found) {
+        entities.push(found);
+      } else {
+        // missing or entity from other type
+        try {
+          const type = Entity.extractTypeFromId(s);
+          const entity = await this.entityMapperService.load<E>(type, s);
```

Comment:
Is it worth optimizing here and only try to load missing entities if they are not the configured type?